plugins/org.eclipse.n4js.semver/src/org/eclipse/n4js/semver/SEMVERMatcher.java

```diff
@@ -0,0 +1,375 @@
+package org.eclipse.n4js.semver;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.n4js.semver.SEMVER.NPMVersion;
+import org.eclipse.n4js.semver.SEMVER.SimpleVersion;
+import org.eclipse.n4js.semver.SEMVER.URLVersion;
+import org.eclipse.n4js.semver.SEMVER.VersionComparator;
+import org.eclipse.n4js.semver.SEMVER.VersionNumber;
+import org.eclipse.n4js.semver.SEMVER.VersionPart;
+import org.eclipse.n4js.semver.SEMVER.VersionRange;
+import org.eclipse.n4js.semver.SEMVER.VersionRangeSet;
+
+/** Utility class to provide methods to check matching of versions. */
+public class SEMVERMatcher {
+
+	/** Relation between two {@link VersionNumber}s */
+	public enum VersionNumberRelation {
+		/** Version A is smaller than version B */
+		Smaller,
+		/** Version A is greater than version B */
+		Greater,
+		/** Version A and B are unrelated */
+		Equal,
+		/** Version A and B are unrelated */
+		Unrelated;
```

Comment:
which literal represents the case that two version numbers differ only in the build meta-data?

Reply:
version numbers that differ _only_ in their build-metadata tag are equal regarding all `compare`, `relate` and `match` methods. However, it might be reasonable to introduce sorting in the method `compareLoose` but the question would be how to sort them. Of course, this sorting would be against Semver semantics.